### PR TITLE
Fix rejected pusher test in worker mode

### PR DESCRIPTION
**Pull request by erikjohnston**
It assumed that pusher changes would propagate instantly.

### CODE REVIEW

File: tests/61push/08_rejected_pushers.pl [hunk 1]
```diff
@@ -74,45 +72,50 @@ sub wait_for_push
...
-              uri     => "/r0/pushers",
-         )->then( sub {
-            my ( $body ) = @_;
-
-            assert_json_keys( $body, qw( pushers ) );
-            @{ $body->{pushers} } == 2 or die "Expected two pushers";
-
-            Future->done(1);
-         });
+         retry_until_success {
```
Comment (richvdh) [inline tests/61push/08_rejected_pushers.pl]: why is this needed? Don't the update and the GET both hit the master?

File: tests/61push/08_rejected_pushers.pl [hunk 2]
```diff
@@ -23,17 +23,15 @@ sub wait_for_push
...
-     my $body = $request->body_from_json;
-
-     return unless $body->{notification}{type};
-     return unless $body->{notification}{type} eq "m.room.message";
-     return 1;
-   })->then( sub {
       my ( $request ) = @_;
+      my $body = $request->body_from_json;
 
+      # Respond to all requests, even if we filiter them out
```
Comment (richvdh) [inline tests/61push/08_rejected_pushers.pl]:
```suggestion
      # Respond to all requests, even if we filter them out
```

File: tests/61push/08_rejected_pushers.pl (hunk 1 above)
Comment (erikjohnston) [reply tests/61push/08_rejected_pushers.pl]: Currently, yes, but there's little reason for that so I figured I may as well change it to not assume that while I'm there anyway.

File: tests/61push/08_rejected_pushers.pl (hunk 2 above)
Comment (erikjohnston) [reply tests/61push/08_rejected_pushers.pl]: Where has my spell checker gone :/

Comment (richvdh) [reply tests/61push/08_rejected_pushers.pl]: apparently he lives in Cambridge :angry:
